actions/mongodb_query_admincommand_action_test.go

```diff
@@ -18,238 +18,296 @@ package actions
-		expected := []interface{}{"mongod", "--sslMode=requireSSL", "--sslPEMKeyFile=/etc/ssl/certificates/server.pem"}
-
-		var tlsMode bool
-		for _, arg := range argv {
-			if strings.Contains(arg.(string), "tlsMode") {
-				tlsMode = true
-				break
-			}
-		}
-		if tlsMode {
-			expected = []interface{}{"mongod", "--tlsMode", "requireTLS", "--tlsCertificateKeyFile", "/etc/ssl/certificates/server.pem"}
-		}
-		assert.Subset(t, argv, expected)
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, nil, "getCmdLineOpts", 1, createTempDir(t)})
+		getCmdLineOptsAssertionsWithoutAuth(t, b)
+	})
+
+	t.Run("replSetGetStatus", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, nil, "replSetGetStatus", 1, createTempDir(t)})
+		replSetGetStatusAssertionsReplicated(t, b)
+	})
+
+	t.Run("getDiagnosticData", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, nil, "getDiagnosticData", 1, createTempDir(t)})
+		getDiagnosticDataAssertions(t, b)
+	})
+}
+
+func TestMongoDBActionsReplWithSSL(t *testing.T) {
+	t.Parallel()
+
+	dsn, files := tests.GetTestMongoDBReplicatedWithSSLDSN(t, "../")
+
+	t.Run("getParameter", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, files, "getParameter", "*", createTempDir(t)})
+		getParameterAssertions(t, b)
+	})
 
-		assert.Equal(t, "1", m.Get("ok").String())
+	t.Run("buildInfo", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, files, "buildInfo", 1, createTempDir(t)})
+		buildInfoAssertions(t, b)
 	})
+
+	t.Run("getCmdLineOpts", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, files, "getCmdLineOpts", 1, createTempDir(t)})
+		getCmdLineOptsAssertionsWithSSL(t, b)
+	})
+
+	t.Run("replSetGetStatus", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, files, "replSetGetStatus", 1, createTempDir(t)})
+		replSetGetStatusAssertionsReplicated(t, b)
+	})
+
+	t.Run("getDiagnosticData", func(t *testing.T) {
+		t.Parallel()
+		b := runAction(t, &MongoDBQueryAdmincommandActionParams{"", dsn, files, "getDiagnosticData", 1, createTempDir(t)})
+		getDiagnosticDataAssertions(t, b)
+	})
+}
+
+func runAction(t *testing.T, params *MongoDBQueryAdmincommandActionParams) []byte {
+	t.Helper()
+	a := NewMongoDBQueryAdmincommandAction(*params)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	b, err := a.Run(ctx)
+	require.NoError(t, err)
+	return b
+}
+
+func convertToObjxMap(t *testing.T, b []byte) objx.Map {
+	t.Helper()
+	data, err := agentpb.UnmarshalActionQueryResult(b)
+	require.NoError(t, err)
+	t.Log(spew.Sdump(data))
+	assert.Len(t, data, 1)
+	return data[0]
+}
+
+func getParameterAssertions(t *testing.T, b []byte) { //nolint:thelper
```

Comment:
Why you don't want to mark these functions as helpers?

Reply:
I'm not sure are they helpers or tests themself. From docs for helpers: _When printing file and line information, that function will be skipped._ So I decided there is no need to mark them as helpers.

Reply:
Yep, it seems that you're right.